Plugin: BannersEverywhere

lillithrosepup opened the pull request:
# BannersEverywhere

Displays nitro and USRBG banners as a background in the member list

![Adds banners to member list](https://github.com/Vendicated/Vencord/assets/44179559/d8a3a2f2-8491-4a4f-b43e-22fdf46622e5)

TheKodeToad commented on src/plugins/bannersEverywhere/index.css:
```diff
@@ -0,0 +1,14 @@
+/* stylelint-disable property-no-vendor-prefix */
+.vc-banners-everywhere-memberlist {
+    opacity: 0.8;
+    -webkit-mask-image: linear-gradient(to right, transparent 20%, #fff);
```
seems like you stopped needing this some time in 2022?

ImLvna replied:
chrome (or atleast the chrome electron uses) doesnt have mask-image without the vendor prefix

TheKodeToad replied:
yeah it's weird but seems to be true.
https://caniuse.com/css-masks
confusing docs :woozy_face:

Vendicated commented on src/plugins/bannersEverywhere/index.css:
```diff
@@ -0,0 +1,14 @@
+    mask-image: linear-gradient(to right, transparent 20%, #fff);
+    width: 100%;
+    height: 100%;
+    content: "";
```
don't think this is needed, no?

ImLvna replied:
it seems so, iirc that was one of the snippets from autumn's pr that i took

Vendicated commented on src/plugins/bannersEverywhere/index.css:
```diff
@@ -0,0 +1,11 @@
+/* stylelint-disable property-no-vendor-prefix */
+.vc-banners-everywhere-memberlist {
+    opacity: 0.8;
+    -webkit-mask-image: linear-gradient(to right, transparent 20%, #fff);
```
you should also add a non prefix variant. likely necessary on firefox, and possibly latest chromium which removed the prefix
```suggestion
    -webkit-mask-image: linear-gradient(to right, transparent 20%, #fff);
    mask-image: linear-gradient(to right, transparent 20%, #fff);
```

Vendicated commented on src/plugins/bannersEverywhere/README.md:
```diff
@@ -0,0 +1,5 @@
+
+Displays nitro and USRBG banners as a background in the member list
+
+![Adds banners to member list](https://github.com/Vendicated/Vencord/assets/44179559/d8a3a2f2-8491-4a4f-b43e-22fdf46622e5)
```
```suggestion
![](https://github.com/Vendicated/Vencord/assets/44179559/d8a3a2f2-8491-4a4f-b43e-22fdf46622e5)
```

Vendicated commented on src/plugins/bannersEverywhere/index.tsx:
```diff
@@ -0,0 +1,154 @@
+import { RestAPI } from "@webpack/common";
+import { User } from "discord-types/general";
+
+import style from "./index.css?managed";
```
your style does not need to be managed

Vendicated commented on src/plugins/bannersEverywhere/index.tsx:
```diff
@@ -0,0 +1,154 @@
+        default: false
+    },
+});
+const data: { [key: string]: string | null; } = {};
```
```suggestion
const data: Record<string, string | null> = {};
```

Vendicated commented on src/plugins/bannersEverywhere/index.tsx:
```diff
@@ -0,0 +1,154 @@
+            continue;
+        }
+
+        data[
```
why keep your own data collection? get it from and subscribe to the UserProfileStore

you also don't seem to be handling guild profiles at all

ImLvna replied:
> you also don't seem to be handling guild profiles at all, so probably should try to add that

i don't see a utility function to fetch member profiles, only users. how would i go about that?

Vendicated commented on src/plugins/bannersEverywhere/index.tsx:
```diff
@@ -0,0 +1,154 @@
+
+let enabled = false;
+
+async function refreshVisibleMembers() {
```
this entire function is incredibly awful. you should patch the component of a member row and `useStateFromStores([UserProfileStore], ...)` in there

ImLvna replied:
im sorry, i dont understand? this uses dom to see what userlist elements are in the viewport and fetches them, i dont think thats possible from just react side

Vendicated replied:
i was suggesting that instead of doing this, you just patch the memberlist row component to insert your own hook call that subscribes to the user profile store and fetches profile if missing

this will then automatically rerender the component when the user profile has eventually been fetched, and Discord's lazy scroller already makes it so that only visible users will be rendered

like
```js
"$self.useFetchMemberProfile($1)"

useFetchMemberProfile(userId: string) {
	const profile = useStateFromStore([UserProfileStore], () => UserProfileStore.getProfile(userId));
	
	useEffect(() => {
		let cancel = false;

		queue.push(() => {
			if (cancel) return;
			return fetchMember(userId).finally(() => sleep(1000));
		})

		return () => cancel = true;
	}, []);
}
```

Vendicated replied:
hope that makes sense!

thororen1234 committed:
MemberCount: Fix error when not in a channel (#2277)

Vendicated committed:
fix ReviewDB

Vendicated committed:
fix RestAPI find

waresnew committed:
WhoReacted: fix scroll jumping when rendering users (#2271)

Co-authored-by: V <vendicated@riseup.net>

Masterjoona committed:
fix(AnonymiseFileNames): anonymise files in forum posts (#2270)

trwy7 committed:
AlwaysTrust: Add options for each popup (#2234)

Co-authored-by: V <vendicated@riseup.net>

Syncxv committed:
PinDMs: add category support & fix bugs (#2203)

Co-authored-by: V <vendicated@riseup.net>

MrDiamondDog committed:
feat(PatchHelper): Paste Full Patch (#1982)

Inbestigator committed:
new plugin: OverrideForumDefaults (#2272)

Co-authored-by: Vendicated <vendicated@riseup.net>

mcpower committed:
MessageLinkEmbeds: Load embeds from newest to oldest (#2230)

Co-authored-by: V <vendicated@riseup.net>

Vendicated committed:
bump to v1.7.3

Vendicated committed:
Fix CloudSync

Nuckyz committed:
BetterNotes: Fix patches

Nuckyz committed:
UserVoiceShow: Fix UserPopout patch stealing predicates from another component

Nuckyz committed:
Fix broken patches

Nuckyz committed:
Reporter: Ignore another useless error

Vendicated committed:
webpack: fix infinite recursion when using ConsoleShortcuts plugin

Nuckyz committed:
Fix lazy chunk force loading

Vendicated committed:
ReadAllNotificationButton: make button less ugly

Vendicated committed:
Fix WhoReacted

Vendicated committed:
Fix MessageLogger

Vendicated committed:
Fix PinDMs

Vendicated committed:
Fix FakeNitro

Vendicated committed:
i may be stupid :3

AutumnVN committed:
fix ImageZoom, LoadingQuote, NoRPC, RevealAllSpoilers, SecretRingTone (#2306)

TheKodeToad committed:
Fix DisableCallIdle

AutumnVN committed:
fix BetterGifPicker, Experiments

Nuckyz committed:
Fix store finds

Nuckyz committed:
Fix BetterSettings patches (#2307)

Nuckyz committed:
Fix more patches

Nuckyz committed:
ShowHiddenChannels: Fix patches

Nuckyz committed:
FakeNitro: Make Soundboard sounds not require boost level

Nuckyz committed:
Bump to 1.7.4

Nuckyz committed:
Decor: Fix find targetting wrong module in some cases

Syncxv committed:
fix(PinDMs): display properly when there are 0 unpinned dms (#2281)

Co-authored-by: Nuckyz <61953774+Nuckyz@users.noreply.github.com>
Co-authored-by: V <vendicated@riseup.net>

Vendicated committed:
fix minor updater bugs

Vendicated committed:
FakeNitro: Fix sending unavailable emotes of the current server

Kodarru committed:
ReviewDB: Fix website url (#2318)

Co-authored-by: V <vendicated@riseup.net>

Nuckyz committed:
ResurrectHome: Fix Force Server Home patch

Kyuuhachi committed:
fix(MessageLinkEmbeds): Actually disable when disabled (#2323)

Nuckyz committed:
Fix MutualGroupDMs and UserVoiceShow patch

puppy-girl committed:
PinDMs: Fix unexpected behaviours when using last message sort (#2324)

Vendicated committed:
ShowHiddenChannels: Fix incompatibility with favorite channels experiment

waresnew committed:
Better error for primitives on proxyLazy + fix StartupTimings (#2339)

AutumnVN committed:
plugin settings: fix filter dropdown having no padding (#2332)

byeoon committed:
EmoteCloner: fix low quality; don't count managed emojis (#2321)

Co-authored-by: Nam Anh <phamnamanh25@gmail.com>
Co-authored-by: ryan-0324 <77452312+ryan-0324@users.noreply.github.com>
Co-authored-by: V <vendicated@riseup.net>

Sqaaakoi committed:
TypingIndicator: Add an option to show user avatars (#2319)

Co-authored-by: Vendicated <vendicated@riseup.net>

Vendicated committed:
FriendsSince: Add icon to be consistent with "member since"

Co-authored-by: Trey <47907719+trwy7@users.noreply.github.com>

nakoyasha committed:
new plugin UnlockedAvatarZoom: allows crop zooming in further (#2287)

Co-authored-by: V <vendicated@riseup.net>

AutumnVN committed:
NoMosaic: make loading image work with responsive layout (#2095)

Vendicated committed:
bump to v1.7.5

Nuckyz committed:
Array support for find + ResurrectHome: View Server Home Button on Server Guide (#2283)

Vendicated committed:
partially revert "Array support for find + ResurrectHome: View Server Home Button on Server Guide (#2283)"

This reverts commit 5636f9d9797bf64170db56447f4bf38f8fadf169.

It breaks Vesktop which is not acceptable. Need to resolve this conflict and add this back later

Vendicated committed:
Fix SpotifyControls

sillyElvy committed:
fix ShowMeYourName (#2354)

AutumnVN committed:
viewIcons: fix for banners (#2351)

Vendicated committed:
fix pronoundb

Vendicated committed:
fix extractAndLoadChunks for when there are no chunks to be loaded

Vendicated committed:
bump to v1.7.6

Vendicated committed:
fix(FavoriteEmojiFirst): don't lower suggestion count on hover

AutumnVN committed:
fix: RoleColorEverywhere, PictureInPicture, NoMosaic (#2356)

sillyElvy committed:
fix MessageLogger (#2358)

Vendicated committed:
fix ShowConnections

Vendicated committed:
bump to v1.7.7

Vendicated committed:
fix settings patch

Vendicated committed:
ShowConnections: fix icon theme logic

Vendicated committed:
CustomRPC: fix preview styles

Vendicated committed:
bump to v1.7.8

sillyElvy committed:
fix ValidUser (#2369)

Vendicated committed:
RoleColorEverywhere: fix chat mentions

Vendicated committed:
ValidUser: fix not working for @unknown-user mentions

Vendicated committed:
fix FriendsSince on canary

Vendicated committed:
Make Updater slightly more future proof

- Removes the option to disable update notifications. Users really should not be outdated, so this option was never good. To disable notifications, turn on auto update
- Enables auto update by default. Users keep complaining about issues while being outdated, so this should help
- Update Notification now opens Updater in a modal to remove dependency on Settings patch. This makes it slightly more failsafe, it's unlikely that both modals and our settings patch break

dolfies committed:
ShowTimeouts->ShowHiddenThings ~show invite-disabled tooltip too (#2375)

aamiaa committed:
New Plugin: BetterSessions (#1324)

Co-authored-by: V <vendicated@riseup.net>

Vendicated committed:
fix FakeProfileThemes

Vendicated committed:
fix badges

My-Name-Is-Jeff committed:
fix ValidUser (#2381)

dolfies committed:
feat(plugin): ImplicitRelationships (#947)

Co-authored-by: Angelos Bouklis <53124886+ArjixWasTaken@users.noreply.github.com>
Co-authored-by: V <vendicated@riseup.net>

Kodarru committed:
feat(plugin): StreamerModeOnStream (#2320)

Co-authored-by: V <vendicated@riseup.net>

sillyElvy committed:
Fix OnePingPerDM and RoleColorEverywhere patches (#2387)

Nuckyz committed:
PronounDB: Fix patch

Nuckyz committed:
MoreUserTags: Fix patches

Nuckyz committed:
FriendsSince: Remove workaround for stable compatibility

Nuckyz committed:
RoleColorEverywhere: Fix patch again

Nuckyz committed:
Bump to 1.7.9

dolfies committed:
fix(ImplicitRelationships): Use new Flux event (#2392)

Nuckyz committed:
BetterFolders: Fix patch; MessageLogger: Ignore Venbot; ReviewDB: Clean migration code

HAHALOSAH committed:
fix(fakeNitro): Use getEmojiURL as emoji.url was removed (#2401)

Nuckyz committed:
Bump to 1.8.0

Nuckyz committed:
RoleColorEverywhere: Fix patch

Nuckyz committed:
ResurrectHome: Fix patch

Nuckyz committed:
RoleColorEverywhere: Actually fix patch

Nuckyz committed:
NoTrack: Option to keep analytics, improve patches

Vendicated committed:
improve contributor modal & badge

Vendicated committed:
fix FriendsSince

Nuckyz committed:
refactor(Webpack): more reliable patching (#2237)

HAHALOSAH committed:
messageLogger: fix niche bug ignoring edits when content is same (#2403)

dolfies committed:
feat(plugin): PauseInvitesForever (#2372)

Co-authored-by: V <vendicated@riseup.net>

Nuckyz committed:
MoreUserTags: Remove old workaround; MessageClickActions: Move finds outside of start

dolfies committed:
fix sort conflict of ImplicitRelationships & SortFriendRequests (#2408)

Sqaaakoi committed:
ReviewDB: add more context menu shortcuts to view reviews (#2382)

Co-authored-by: V <vendicated@riseup.net>

katlyn committed:
feat(USRBG): update to new API (#2388)

sunnniee committed:
MsgClickActions: control ping via shift & NoReplyMention plugin (#2390)

Co-authored-by: V <vendicated@riseup.net>

Vendicated committed:
fix badges with custom component

Kyuuhachi committed:
feat(plugin): ImageLink (#2297)

Nuckyz committed:
Test Patches: Faster chromium setup; Update action versions (#2412)

Nuckyz committed:
Test Patches: Remove pnpm add puppeteer as it's already in deps

Nuckyz committed:
SecretRingToneEnabler: Fix patch

dolfies committed:
feat(ValidUser): also display badges & banner (#2235)

Co-authored-by: V <vendicated@riseup.net>

GabiRP committed:
NewGuildSettings: add push notifs, highlights & events (#2413)

Co-authored-by: V <vendicated@riseup.net>

Vendicated committed:
fix overflow when having two rows of badges

Nuckyz committed:
Future proof FakeNitro and Experiments to not brick Discord startup

Vendicated committed:
bump to v1.8.1

Nuckyz committed:
Resolve PluginSettings circular deps better

Nuckyz committed:
ImplicitRelationships: Properly test find

Faf4a committed:
ReviewDB: Fix context menus being added to folders (#2416)

Sqaaakoi committed:
ImageLink: Fix embed showing in gifs (#2417)

Vendicated committed:
fix webpack patch on client using discordapp.com

Vendicated committed:
fix webpack patch on client using discordapp.com part 2

Nuckyz committed:
FakeNitro: Make disableEmbedPermissionCheck setting not private

Vendicated committed:
upgrade nodejs & pnpm to latest versions

Nuckyz committed:
Fix subscribing to plugin flux events twice

Nuckyz committed:
FakeNitro: Add custom notifications sound bypass

Vendicated committed:
fix useStateFromStores JSDoc

Co-authored-by: Nuckyz <61953774+Nuckyz@users.noreply.github.com>

Vendicated committed:
Revert "messageLogger: fix niche bug ignoring edits when content is same (#2403)"

This reverts commit 85d6d74a3e14ed4a1cc83c8131b83c08cfdce04a.

As suspected, this code was actually necessary

Nuckyz committed:
BetterFolders: Fix broken patch

Vendicated committed:
improve SupportHelper

catgirlcataclysm committed:
new plugin WebScreenShareFixes: remove low stream bitrate limit (#2405)

arHSM committed:
new plugin MessageLatency: indicator for other people's latency (#2353)

Kyuuhachi committed:
new plugin ReplyTimestamp: show timestamps of replied messages  (#2296)

Co-authored-by: vee <vendicated@riseup.net>

Commit:
feat(plugin): VoiceDownload (#2280)

Co-authored-by: vee <vendicated@riseup.net>

dolfies committed:
showHiddenThings: also show ModView & hidden discovery servers (#2415)

Co-authored-by: vee <vendicated@riseup.net>

KK2-5 committed:
LastfmRichPresence: Add option to use album name as status name (#2400)

Co-authored-by: vee <vendicated@riseup.net>

Vendicated committed:
bump to v1.8.2

Nuckyz committed:
BetterFolders: Fix component erroring

Nuckyz committed:
CrashHandler: Increment timeout for trying to recover

Vendicated committed:
fix(BetterSettings): do not catch errors of other ui

Vendicated committed:
VoiceDownload: fix doing nothing on discord desktop app

Vendicated committed:
fix(SupportHelper): dont flag vencord web as externally updated

Vendicated committed:
BetterSettings: fix error handling crashing in some niche cases

Vendicated committed:
new plugin ShowTimeoutDuration ~ shows how much longer a user's timeout will last

Vendicated committed:
im the dumbest dumdum

Nuckyz committed:
ErrorBoundary some more components

shadowpercifal committed:
permissionsViewer: add role & user context menus to copy id (#2436)

Co-authored-by: V <vendicated@riseup.net>

nyakowint committed:
XSOverlay: add settings for different notification types (#2055)

Co-authored-by: vee <vendicated@riseup.net>

arHSM committed:
messageLatency: fix grammar & add aliucord/kotlin client tooltip (#2426)

ImBonana committed:
SilentTyping: add chat input context menu option to toggle (#2386)

Co-authored-by: vee <vendicated@riseup.net>

FractalGeodes committed:
ThemeAttributes: add data-author-username to messages (#2422)

nin0-dev committed:
CustomRPC: show warning when game activity is disabled (#2245)

Co-authored-by: V <vendicated@riseup.net>

Nuckyz committed:
FakeNitro: Update description

Faf4a committed:
Dearrow: allow configuring which elements get dearrowd (#2414)

Co-authored-by: Vendicated <vendicated@riseup.net>

sillyElvy committed:
pronounDB: Update to API v2 (#2355)

Co-authored-by: vee <vendicated@riseup.net>

Vendicated committed:
settingsSync: include date in filename for better sorting

Co-authored-by: cd CreepArghhh_ <65649991+cd-CreepArghhh@users.noreply.github.com>

Commit:
add Native settings implementation (#2346)

Co-authored-by: vee <vendicated@riseup.net>

HAHALOSAH committed:
Plugin Settings: fix text overflow for long plugin names (#2383)

Co-authored-by: V <vendicated@riseup.net>

Cats1337 committed:
feat(Translate): add toggle for chat bar icon (#2418)

Vendicated committed:
but here's the bumper (v1.8.3)

nyakowint committed:
XSOverlay: Adjust message length timeout (#2445)

dolfies committed:
fix(MessageLatency): Adjust for Discord kotlin clients (#2443)

Nuckyz committed:
PinDMs: ErrorBoundary renderChannel

Vendicated committed:
ValidUser: fix crashing when viewing a valid-userd staff's profile

Vendicated committed:
ValidUser: fix rendering old mentions when message is edited

Fixes https://github.com/Vendicated/Vencord/issues/2451

dolfies committed:
feat(ShowHiddenThings): Remove Discovery banned/NSFW filters (#2453)

axiand committed:
EmoteCloner: allow cloning from reactions (#2458)

aamiaa committed:
fix(MutualGroupDMs): properly pass props (#2457)

Co-authored-by: vee <vendicated@riseup.net>

nakoyasha committed:
FakeNitro: allow using subscription-locked emojis (#2456)

Co-authored-by: Vendicated <vendicated@riseup.net>

Nuckyz committed:
Properly ErrorBoundary recent changes

AnubisNekhet committed:
MessageLogger: use discord variables instead of hardcoded colors (#2428)

Nuckyz committed:
Make all RestAPI calls use Endpoints object

Vendicated committed:
fix settings ui on canary

Vendicated committed:
bump to v1.8.4

Vendicated committed:
fix Vencord Settings section being added multiple times

BoardTM committed:
ThemeAttributes: add larger avatar url variables to avatars (#2449)

Co-authored-by: vee <vendicated@riseup.net>

UlyssesZh committed:
new plugin NoServerEmoji ~ hides server emojis from autocomplete (#1787)

Co-authored-by: vee <vendicated@riseup.net>

UlyssesZh committed:
new plugin CtrlEnterSend (#1794)

Co-authored-by: vee <vendicated@riseup.net>

eepyfemboi committed:
MoreUserTags: Add chat moderator tag (#2424)

Co-authored-by: Vendicated <vendicated@riseup.net>

Nuckyz committed:
Fix BetterSettings & StartupTimings patch

Vendicated committed:
Settings: remove obsolete patch; add redundancy & more useful dbg copy

mcpower committed:
chore: tidy up suggested vscode extensions list (#2221)

Co-authored-by: Vendicated <vendicated@riseup.net>

Nuckyz committed:
BetterFolders: Fix scrolling

waresnew committed:
new plugin ValidReply ~ fix "Message could not be loaded" (#2337)

Co-authored-by: V <vendicated@riseup.net>

Aztup committed:
feat(plugins/openInApp) Add tidal support (#2404)

Co-authored-by: vee <vendicated@riseup.net>

ScattrdBlade committed:
PetPet: Fix Upload Image Option (#2461)

Nuckyz committed:
feat: Allow finds to use regex (#2452)

Nuckyz committed:
Fix: Canonicalize regex finds

waresnew committed:
Fix: Plugins without start/stop function failing to stop/start (#2463)

Nuckyz committed:
extractAndLoadChunksLazy: Cache result to avoid searching factories everytime

rozbrajaczpoziomow committed:
feat(MessageLatency): Show milliseconds option (#2454)

Nuckyz committed:
MessageLatency: Fix wrong constant & false positive

Sqaaakoi committed:
ShowTimeoutDuration: Simplify tooltip style, allow changing style without reload (#2441)

shadowpercifal committed:
fix(readAllNotificationsButton): Mark threads as read (#2437)

verticalsync committed:
feat(ViewIcons): Group & User DMs icons support (#2464)

Nuckyz committed:
ViewIcon: Replace regex find with string find

Nuckyz committed:
FakeNitro: Fix & rewrite emoji bypass patches

JohnyTheCarrot committed:
new plugin AutomodContext (#2290)

ImpishMoxxie committed:
new plugin ReplaceGoogleSearch (#2450)

Nuckyz committed:
FakeNitro: Fix attempting to bypass unicode emojis

Closes #2470

Nuckyz committed:
chore: Fix non standard plugin names

D3SOX committed:
new plugin NoDefaultHangStatus (#2468)

Nuckyz committed:
Add missing README to new plugins

D3SOX committed:
fix(replaceGoogleSearch): correct GitHub casing (#2471)

Yumiro committed:
lastfmRPC: add setting to toggle "View Song" button (#2292)

Co-authored-by: vee <vendicated@riseup.net>

Vendicated committed:
add package for publishing types to npm (#2473)

https://www.npmjs.com/package/@vencord/types

Nuckyz committed:
ViewIcons: Fix Group Icons being clickable in channel list

Nuckyz committed:
ResurrectHome: Likely fix breaking latest messages in chat

Nuckyz committed:
FriendsSince: Show in user profile modal

waresnew committed:
new plugin CustomIdle (#2342)

Noxillio committed:
MoreUserTags: If server owner tag is disabled, do not give other tags (#2219)

waresnew committed:
Fix: PatchHelper not auto filling match field (#2338)

Nuckyz committed:
fix(PatchHelper): Make find and match more responsive

Nuckyz committed:
chore: Make package manager version not strict

Nuckyz committed:
MessageLinkEmbeds: Add limit for nested links

Nuckyz committed:
MessageLinkEmbeds: No longer need to reset global regex

Nuckyz committed:
Change duplicate find for SHC and VCDoubleClick

Nuckyz committed:
Fix: Ignore bundled lib webpack on web

Nuckyz committed:
StartupTimings: Fix patch

Vendicated committed:
bump to v1.8.5

lexisother committed:
fakeProfileThemes: settings UI improvements (#966)

Co-authored-by: V <vendicated@riseup.net>

schoeneBiene committed:
feat(BetterRoleContext): Add option to view role icons (#2482)

Co-authored-by: vee <vendicated@riseup.net>

dolfies committed:
ShowHiddenThings: more effectively explode Algolia filters (#2484)

Co-authored-by: vee <vendicated@riseup.net>

k26pl committed:
fix(MessageLogger): correctly blur spoilered images (#2433)

Co-authored-by: vee <vendicated@riseup.net>

pwall2222 committed:
QuickReply: skip blocked messages if NoBlockedMessages enabled (#2476)

Nonstopgaming75 commented:
pls fix it

Nuckyz committed:
Revert "MessageLinkEmbeds: No longer need to reset global regex"

It is still needed for messageLinkRegex.test

Nuckyz committed:
Fix reporter false positive and DefaultExtractAndLoadChunksRegex not catching all cases

Nuckyz committed:
Fix not setting property on originalOnChunksLoaded

Vendicated committed:
MessageLogger: fix edit logging

Vendicated committed:
fix AnonymiseFilenames

Vendicated committed:
work around discord unloading in background

Nuckyz committed:
Revert removal of DevTools context menu fix

Vendicated committed:
discord why tf would u roll back to 10 days old build???

Vendicated committed:
Bump to v1.8.6

Nuckyz committed:
Optimize slowest patches

Nuckyz committed:
ConsoleShortcuts: Start at Init

mantikafasi committed:
new plugin Summaries: show Discords AI-generated convo summaries (#2481)

Co-authored-by: Vendicated <vendicated@riseup.net>

Vendicated committed:
fix minor bugs in various plugins

- FriendsSince: Don't show for friend requests
- FakeNitro: Fix attempting to bypass unicode emojis #2503
- MessageLatency: ignore bots #2504
- CtrlEnterSend: use cmd+enter on macOS #2502
- ReplaceGoogleSearch: trim LF character #2488

Co-authored-by: AutumnVN <autumnvnchino@gmail.com>
Co-authored-by: rushiiMachine <33725716+rushiiMachine@users.noreply.github.com>
Co-authored-by: Lumap <lumap@duck.com>
Co-authored-by: Mylloon <kennel.anri@tutanota.com>

Vendicated committed:
build: improve fileInclude plugin

lillithrosepup committed:
new plugin WatchTogetherAdblock: block ads in youtube activity (#2021)

Co-authored-by: vee <vendicated@riseup.net>

Nuckyz committed:
Summaries: Fix start error if no summaries-data exist

UlyssesZh committed:
LoadingQuotes: more customization & custom quotes support (#1795)

Co-authored-by: lewisakura <lewi@lewisakura.moe>
Co-authored-by: Vendicated <vendicated@riseup.net>

lillithrosepup committed:
use store instead of manual fetching

lillithrosepup committed:
fix linting

Nonstopgaming75 commented:
It works but I saw you removed Userbg support

Vendicated committed:
Merge branch 'dev' into bannersEverywhere

lillithrosepup committed:
add back in USRBG support

ImLvna commented:
added back usrbg support! only fetches the user profile if usrbg.getImageUrl returns null

lillithrosepup committed:
fix: dont animate banners if the setting is off

ImLvna commented:
not using usrbg banners if the user does not want animated banners, as there is no way of knowing if a usrbg banner is animated or not

lillithrosepup committed:
fix: HEAD usrbg urls

lillithrosepup committed:
fix: lint

lillithrosepup committed:
dont manage the style

lillithrosepup committed:
fix lint

ImLvna commented:
confirmed working after doomsday!
![image](https://github.com/Vendicated/Vencord/assets/44179559/2bf8812c-7326-44a4-9f7e-ebfe8aa1c1ab)

lillithrosepup committed:
Merge branch 'dev' into bannersEverywhere
